lib/project_types/script/layers/infrastructure/languages/typescript_task_runner.rb

```diff
@@ -27,13 +29,18 @@ def compiled_type
           end
 
           def install_dependencies
-            check_node_version!
+            check_system_dependencies!
 
             output, status = ctx.capture2e("npm install --no-audit --no-optional --legacy-peer-deps --loglevel error")
-            raise Errors::DependencyInstallError, output unless status.success?
+            raise Errors::DependencyInstallationError, output unless status.success?
```

Comment:
I think this method should be in `TaskRunner`, and not in its subclasses, and `check_system_dependencies` should be `protected` as well.  Now that I am looking at `check_system_dependencies!`, an alternative approach is to just have an array:
```
tool_req = [{"tool_name": "npm", "min_version": MIN_NPM_VERSION}, {"tool_name": "node", "min_version": MIN_NODE_VERSION}]
```
Then, `check_system_dependencies!` can be implemented in `TaskRunner` to just iterate through this array and call `check_tool_version!`.  That will avoid having this awkward `check_system_dependencies!` throwing `NotImplementedError`.